controllers/awsmachine_controller.go

```diff
@@ -321,13 +325,29 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 	// TODO(vincepri): Remove this annotation when clusterctl is no longer relevant.
 	machineScope.SetAnnotation("cluster-api-provider-aws", "true")
 
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-lifecycle.html
 	switch instance.State {
+	case infrav1.InstanceStatePending:
+		machineScope.V(1).Info("EC2 instance is pending", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 	case infrav1.InstanceStateRunning:
-		machineScope.Info("Machine instance is running", "instance-id", *machineScope.GetInstanceID())
+		machineScope.V(1).Info("EC2 instance is running", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 		machineScope.SetReady()
-	case infrav1.InstanceStatePending:
-		machineScope.Info("Machine instance is pending", "instance-id", *machineScope.GetInstanceID())
-	default:
+	case infrav1.InstanceStateStopping:
+		machineScope.V(1).Info("EC2 instance is stopping", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		machineScope.SetUnready()
+	case infrav1.InstanceStateStopped:
+		machineScope.V(1).Info("EC2 instance is stopped", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		machineScope.SetUnready()
+
+		// We do not error yet so that the final terminated state is captured.
+	case infrav1.InstanceStateShuttingDown:
+		machineScope.V(1).Info("Unexpected EC2 instance termination", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		machineScope.SetUnready()
+		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstanceUnexpectedTermination", "Unexpected instance termination")
+	case infrav1.InstanceStateTerminated:
+		machineScope.V(1).Info("Unexpected EC2 instance termination", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
+		machineScope.SetUnready()
+		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstanceUnexpectedTermination", "Unexpected instance termination")
```

Comment:
These two cases could be simplified to a single one capturing both

Reply:
These ones can't. Not erroring in the case of InstanceStateShuttingDown, because our state machine can then no longer transition to Terminated, hence the comment.